spec/services/update_subscription_address_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'spec_helper'
+
+describe UpdateSubscriptionAddress do
+
+  context "when updating a subscription" do
+    let(:user) { create(:user) }
+    let(:old_address) { create(:subscription_address, address1: "Old Address", user: user) }
+    let(:subscription) { create(:subscription, ship_address: old_address, bill_address: old_address, user: user) }
+
+    before do
+      @new_address = build_address_params(old_address)
+      @new_address[:address1] = "New Address"
+    end
```

Comment:
ah interesting... I would do the opposite:  test for all the fields in this test and test only one in your feature test to validate that the update request is handled properly.